engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/UserTaskTest.java

```diff
@@ -418,6 +420,191 @@ public void shouldCreateJobWithEvaluatedCandidateUsersExpressionHeader() {
         .containsEntry(Protocol.USER_TASK_CANDIDATE_USERS_HEADER_NAME, "[\"jack\",\"rose\"]");
   }
 
+  @Test
+  public void shouldCreateJobWithDueDateHeader() {
+    // given
+    ENGINE.deployment().withXmlResource(process(t -> t.zeebeDueDate(DUE_DATE))).deploy();
+
+    // when
+    final long processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    // then
+    final Record<JobRecordValue> job =
+        RecordingExporter.jobRecords(JobIntent.CREATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .getFirst();
+
+    final Map<String, String> customHeaders = job.getValue().getCustomHeaders();
+    assertThat(customHeaders)
+        .hasSize(1)
+        .containsEntry(Protocol.USER_TASK_DUE_DATE_HEADER_NAME, DUE_DATE);
+  }
+
+  @Test
+  public void shouldCreateJobWithEvaluatedDueDateExpressionHeader() {
+    // given
+    ENGINE.deployment().withXmlResource(process(t -> t.zeebeDueDateExpression("dueDate"))).deploy();
+
+    // when
+    final long processInstanceKey =
+        ENGINE
+            .processInstance()
+            .ofBpmnProcessId(PROCESS_ID)
+            .withVariables(Map.of("dueDate", DUE_DATE))
+            .create();
+
+    // then
+    final Record<JobRecordValue> job =
+        RecordingExporter.jobRecords(JobIntent.CREATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .getFirst();
+
+    final Map<String, String> customHeaders = job.getValue().getCustomHeaders();
+    assertThat(customHeaders)
+        .hasSize(1)
+        .containsEntry(Protocol.USER_TASK_DUE_DATE_HEADER_NAME, DUE_DATE);
+  }
+
+  @Test
+  public void shouldCreateJobAndIgnoreEmptyDueDateHeader() {
```

Comment:
:wrench: What about creating test cases for null `dueDate` and `followUpDate` values? We can also include them in existing tests if that will be simpler.

Reply:
Thanks for that! I'll create a separate test case since the `ResultType` of the `EvaluationResult` will be different, so we will be testing different code.